pkg/utils/report/results.go

```diff
@@ -86,72 +87,74 @@ func SeverityFromString(severity string) policyreportv1alpha2.PolicySeverity {
-			result := policyreportv1alpha2.PolicyReportResult{
-				Source:  kyverno.ValueKyvernoApp,
-				Policy:  key,
-				Rule:    ruleResult.Name(),
-				Message: ruleResult.Message(),
-				Result:  toPolicyResult(ruleResult.Status()),
-				Scored:  annotations[kyverno.AnnotationPolicyScored] != "false",
-				Timestamp: metav1.Timestamp{
-					Seconds: time.Now().Unix(),
-				},
-				Category: annotations[kyverno.AnnotationPolicyCategory],
-				Severity: SeverityFromString(annotations[kyverno.AnnotationPolicySeverity]),
-			}
-			if ruleResult.Exception() != nil {
-				result.Properties = map[string]string{
-					"exception": ruleResult.Exception().Name,
-				}
-			}
-			pss := ruleResult.PodSecurityChecks()
-			if pss != nil {
-				var controls []string
-				for _, check := range pss.Checks {
-					if !check.CheckResult.Allowed {
-						controls = append(controls, check.ID)
-					}
-				}
-				if len(controls) > 0 {
-					sort.Strings(controls)
-					result.Properties = map[string]string{
-						"standard": string(pss.Level),
-						"version":  pss.Version,
-						"controls": strings.Join(controls, ","),
-					}
-				}
-			}
-			if result.Result == "fail" && !result.Scored {
-				result.Result = "warn"
+func ToPolicyReportResult(policyType engineapi.PolicyType, policyName string, ruleResult engineapi.RuleResponse, annotations map[string]string, resource *corev1.ObjectReference) policyreportv1alpha2.PolicyReportResult {
+	result := policyreportv1alpha2.PolicyReportResult{
+		Source:  kyverno.ValueKyvernoApp,
+		Policy:  policyName,
+		Rule:    ruleResult.Name(),
+		Message: ruleResult.Message(),
+		Result:  toPolicyResult(ruleResult.Status()),
+		Scored:  annotations[kyverno.AnnotationPolicyScored] != "false",
+		Timestamp: metav1.Timestamp{
+			Seconds: time.Now().Unix(),
+		},
+		Category: annotations[kyverno.AnnotationPolicyCategory],
+		Severity: SeverityFromString(annotations[kyverno.AnnotationPolicySeverity]),
+	}
+	if result.Result == "fail" && !result.Scored {
+		result.Result = "warn"
+	}
+	if resource != nil {
+		result.Resources = []corev1.ObjectReference{
+			*resource,
+		}
+	}
+	if ruleResult.Exception() != nil {
+		result.Properties = map[string]string{
+			"exception": ruleResult.Exception().Name,
+		}
+	}
+	pss := ruleResult.PodSecurityChecks()
+	if pss != nil && len(pss.Checks) > 0 {
+		var controls []string
+		for _, check := range pss.Checks {
+			if !check.CheckResult.Allowed {
+				controls = append(controls, check.ID)
 			}
-			results = append(results, result)
 		}
-	} else {
-		for _, ruleResult := range response.PolicyResponse.Rules {
-			result := policyreportv1alpha2.PolicyReportResult{
-				Source:  "ValidatingAdmissionPolicy",
-				Policy:  ruleResult.Name(),
-				Message: ruleResult.Message(),
-				Result:  toPolicyResult(ruleResult.Status()),
-				Timestamp: metav1.Timestamp{
-					Seconds: time.Now().Unix(),
-				},
+		if len(controls) > 0 {
+			sort.Strings(controls)
+			result.Properties = map[string]string{
```

Comment:
This could potentially replace properties set on line 113. May be best to create an empty map for properties in the declaration on line 91, and then add properties to that for the exception and PSS controls.

Reply:
@JimBugwadia i have updated that, please have a look